src/faction_camp.cpp

```diff
@@ -4464,72 +4464,77 @@ bool basecamp::survey_field_return( const mission_id &miss_id )
     }
 
     popup( _( "Select a tile up to %d tiles away." ), 1 );
-    const tripoint_abs_omt where( ui::omap::choose_point() );
-    if( where == overmap::invalid_tripoint ) {
-        return false;
-    }
 
-    int dist = rl_dist( where.xy(), omt_pos.xy() );
-    if( dist != 1 ) {
-        popup( _( "You must select a tile within %d range of the camp" ), 1 );
-        return false;
-    }
-    if( omt_pos.z() != where.z() ) {
-        popup( _( "Expansions must be on the same level as the camp" ) );
-        return false;
-    }
-    const point dir = talk_function::om_simple_dir( omt_pos, where );
-    if( expansions.find( dir ) != expansions.end() ) {
-        if( query_yn(
-                _( "You already have an expansion at that location.  Do you want to finish this mission?  If not, the mission remains active and another tile can be checked." ) ) ) {
-            finish_return( *comp, true, abort_msg, skill_construction.str(), 0 );
-            return true;
-        } else {
+    tripoint_abs_omt where( get_player_character().global_omt_location() );
+
+    while( true ) {
+        where = ui::omap::choose_point( where );
+        if( where == overmap::invalid_tripoint ) {
             return false;
         }
```

Comment:
The `return false` path is the way out of the loop without changing any state and is triggered by pressing escape while selecting a tile.

Reply:
Is that stated in the UI?

Reply:
#77397 includes a feature to put a message/instructions at the top of the map tile choosing interface. If that would make this feel better, I can break it out as its own PR and get it merged so this PR can use it.